rust-runtime/aws-smithy-http-server/src/plugin/alb_health_check.rs

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Middleware for handling [ALB health
+//! checks](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html).
+//!
+//! # Example
+//!
+//! ```no_run
+//! # use aws_smithy_http_server::{body, plugin::{PluginPipeline, alb_health_check::AlbHealthCheckLayer}};
+//! # use hyper::{Body, Response, StatusCode};
+//! let plugins = PluginPipeline::new()
+//!     // Handle all `/ping` health check requests by returning a `200 OK`.
+//!     .http_layer(AlbHealthCheckLayer::new("/ping", |_req| async {
+//!         Response::builder()
+//!             .status(StatusCode::OK)
+//!             .body(body::boxed(Body::empty()))
+//!             .expect("Couldn't construct response")
+//!     }));
+//!
+//! ```
+
+use std::convert::Infallible;
+use std::task::{Context, Poll};
+
+use futures_util::{Future, FutureExt};
+use http::StatusCode;
+use hyper::{Body, Request, Response};
+use pin_project_lite::pin_project;
+use tower::{service_fn, util::Oneshot, Layer, Service, ServiceExt};
+
+use crate::body::BoxBody;
+
+use super::either::EitherProj;
+use super::Either;
+
+/// A [`tower::Layer`] used to apply [`AlbHealthCheckService`].
+#[derive(Clone, Debug)]
+pub struct AlbHealthCheckLayer<'a, HealthCheckHandler> {
+    health_check_uri: &'a str,
```

Comment:
I don't think this will work in any case other than `'static`, this probably needs to be a `Cow<'static, str>`.

Reply:
I think you're right, but this can be a `Cow<'a, str>`, right?

Reply:
If `&'a str` doesn't work I'd expect `Cow::<'a, str>::Borrow` to not work either. Lemme try it in the `pokemon-service` quickly.

Reply:
Yeah
```rust
    let ping = "/ping".to_string();

/* ... */

        .http_layer(AlbHealthCheckLayer::from_handler(&ping, |_req| async {
            StatusCode::OK
        }));
```

yields

```
error[E0597]: `ping` does not live long enough
  --> pokemon-service/src/main.rs:57:55
   |
57 |           .http_layer(AlbHealthCheckLayer::from_handler(&ping, |_req| async {
   |                                                         ^^^^^ borrowed value does not live long enough
...
61 |       let app = PokemonService::builder_with_plugins(plugins)
   |  _______________-
62 | |         // Build a registry containing implementations to all the operations in the service. These
63 | |         // are async functions or async closures that take as input the operation's input and
64 | |         // return the operation's output.
65 | |         .get_pokemon_species(get_pokemon_species)
   | |_________________________________________________- argument requires that `ping` is borrowed for `'static`
...
94 |   }
   |   - `ping` dropped here while still borrowed
```

Reply:
Yeah, you're totally right. Done in d56d5cd46.